WcaOnRails/app/models/registration.rb

```diff
@@ -102,7 +107,7 @@ def waiting_list_info
 
   validate :must_register_for_gte_one_event
   private def must_register_for_gte_one_event
-    if events.empty?
+    if registration_events.length { |r| !r.marked_for_destruction? } == 0
```

Comment:
Good catch guys. I've created #883, #884, and #885 to track the other similar issues in our codebase. I found them all by looking for occurrences of `accepts_nested_attributes_for` in our codebase.